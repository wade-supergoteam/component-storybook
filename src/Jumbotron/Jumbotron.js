import styled, { css } from "styled-components";
import stripUnit from "polished/lib/helpers/stripUnit";
import themeProp from "../utils/theme";

import { borderRadiusLg, jumbotronPadding, jumbotronBg } from "./default-theme";

import { borderRadius } from "../utils/border-radius";

const Jumbotron = styled.div`
  padding: ${themeProp("jumbotronPadding", jumbotronPadding)}
    ${themeProp("jumbotronPadding", `${stripUnit(jumbotronPadding) / 2}rem`)};

  margin-bottom: ${themeProp("jumbotronPadding", jumbotronPadding)};
  background-color: ${themeProp("jumbotronBg", jumbotronBg)};
  ${borderRadius(themeProp("borderRadiusLg", borderRadiusLg))};

  // @include media-breakpoint-up(sm) {
  //   padding: ($jumbotron-padding * 2) $jumbotron-padding;
  // }

  ${props =>
    props.fluid &&
    css`
      padding-right: 0;
      padding-left: 0;
      ${borderRadius(0)};
    `};
`;

export default Jumbotron;
