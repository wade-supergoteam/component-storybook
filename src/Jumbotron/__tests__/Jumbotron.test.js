import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import Jumbotron from "..";

test("Create Default", () => {
  const wrapper = renderer.create(<Jumbotron />);
  expect(wrapper).toMatchSnapshot();
});

test("Fluid Jumbotron", () => {
  const wrapper = renderer.create(<Jumbotron fluid />);
  expect(wrapper).toMatchSnapshot();
});
