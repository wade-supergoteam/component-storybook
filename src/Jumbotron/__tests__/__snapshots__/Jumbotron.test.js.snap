// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Default 1`] = `
<div
  className="sc-bdVaJa hQCgur"
/>
`;

exports[`Fluid Jumbotron 1`] = `
<div
  className="sc-bdVaJa bPsZxt"
/>
`;
